app/controllers/samples_controller.rb

```diff
@@ -188,6 +188,11 @@ def show
     end
   end
 
+  def show_amr_counts
+    @sample = samples_scope.find(params[:id])
```

Comment:
Since you listed show_amr_counts as a read_action (which makes sense), I think you should replace `samples_scope` by `current_power.samples`. As you can see in the definition of samples_scope, it selects just the "updatable" samples (tighter access control appropriate for edit actions), but what we want is instead all the "viewable" samples, which are accessible as `current_power.samples`. If you inspect `app/models/power.rb`, you'll see that the latter expression through the `viewable` method implemented in the Sample model.

Reply:
Sounds good. I'm going to work on rewriting the display end for amr so hopefully I can just get rid of this soon -- will keep it in mind as I'm making my next set of display related changes.

Reply:
Actually never mind. You should remove that line entirely, because `set_sample` is listed as a `before_action` for all `READ_ACTIONS`. So `@sample = samples_scope.find(params[:id])` happens automatically when show_amr_counts is called (just before)